### ISSUE TITLE
Can't correctly mount a directory contains a file using non-ascii characters.

### ISSUE THREAD

**Issue by niw** [docker/machine#1763]
When we mount a directory to docker-machine, when that directory contains at least one file which has non-ascii characters in its name, we partially or entire files are not visible though the volume.
Tested on next environment.

    $ sw_vers -productVersion 
    10.10.5
    $ VBoxManage --version
    5.0.0r101573
    $ docker-machine --version
    docker-machine version 0.4.1 (HEAD)

**Repro step**

On host OS X, run next command.

    $ mkdir test
    $ cd test
    $ touch "🐱"
    $ cat > Dockerfile
    FROM busybox
    VOLUME /test
    ENTRYPOINT ["ls", "-al", "/test"]
    $ docker build -t test .
    $ docker run --rm -v `pwd`:/test test

**Expected behavior**

Shows `Dockerfile` and `🐱`

**Actual behavior**

Only shows `Dockerfile`

    total 8
    drwxr-xr-x    1 default  staff          136 Aug 26  2015 .
    drwxr-xr-x   24 root     root          4096 Aug 26 06:11 ..
    -rw-r--r--    1 default  staff           60 Aug 26  2015 Dockerfile

Looks like we're not mounting directory with `iocharset=utf8` which prevents us to see these files from the container.

**Comment by niw** [comment]
Original conversation on Twitter.
https://twitter.com/upthecyberpunks/status/636468275308728320
cc: @nathanleclaire

**Comment by niw** [comment]
Confirmed that next command solve the issue (Assuming that docker-machine name is `dev`)

    $ docker-machine ssh dev
    $ sudo umount /Users
    $ sudo mount -t vboxsf -o uid=1000,gid=50,iocharset=utf8 Users /Users

Then,

    $ docker run --rm -v `pwd`:/test test
    total 8
    drwxr-xr-x    1 default  staff          136 Aug 26 11:39 .
    drwxr-xr-x   24 root     root          4096 Aug 26 11:39 ..
    -rw-r--r--    1 default  staff           60 Aug 26 11:39 Dockerfile
    -rw-r--r--    1 default  staff            0 Aug 26 10:09 ????

Note that "🐱" shows as "????" but it is because of busybox ls, it's correctly displayed as "🐱" when using `FROM ubuntu`.

So, possible fix might be updating next init script to give `iocharset=utf8` by default.
https://github.com/boot2docker/boot2docker/blob/master/rootfs/rootfs/etc/rc.d/automount-shares#L11

**Comment by niw** [comment]
Next pull request, which is not updated since May, because of unnecessary `mount.vboxsf` line change, can fix this issue.
boot2docker/boot2docker#697

**Comment by ehazlett** [comment]
Yeah looks like that PR is waiting on a change.  Perhaps we could carry that PR if the contributor doesnt' update soon.

**Comment by niw** [comment]
I made a new pull request referenced from here, boot2docker/boot2docker#1044 which addressed the concerns on the original pull request but not updated.
Hopefully that will be merged soon.

**Comment by niw** [comment]
boot2docker/boot2docker#697 is merged now, so probably this issue will be as well.

**Comment by nathanleclaire** [comment]
Nice work @niw !
